src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,167 @@
+import java.util.Scanner;
+
+import duke.tasklist.TaskList;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.Todo;
+
+/**
+ * Storage class to interact with the text file store in ./data.
+ */
+public class Storage {
+    private String cwd = System.getProperty("user.dir");
+    private java.nio.file.Path dataPath;
+    private String dataPathString;
+
+    /**
+     * Constructor.
+     * @param filePath the path to the text file that stores the serialised tasks
+     */
+    public Storage(String filePath) {
+        this.dataPath = java.nio.file.Paths.get(cwd, filePath.split("/"));
+        this.dataPathString = dataPath.toString();
+    }
+
+    /**
+     * Serialises task into a string that is storable in the text file.
+     * @param task the task to be serialise
+     * @return the string of the serialised task
+     */
+    private String serialise(Task task) {
+        String data = "";
+        String isDone = (task.getIsDone() ? "1" : "0");
+        String description = task.getDescription();
+
+        if (task instanceof Todo) {
+            data = String.join(" | ", "T", isDone, description);
+        }
+
+        if (task instanceof Deadline) {
+            Deadline deadline = (Deadline) task;
+            String by = deadline.getBy();
+            data = String.join(" | ", "D", isDone, description, by);
+        }
+
+        if (task instanceof Event) {
+            Event event = (Event) task;
+            String from = event.getFrom();
+            String to = event.getTo();
+            data = String.join(" | ", "E", isDone, description, from, to);
+        }
+
+        return data;
+    }
+
+    /**
+     * Deserialises the task string to a Task instance.
+     * @param data the serialised task
+     * @return the task that has been deserialised from the input
+     */
+    private Task deserialise(String data) {
+        String[] params = data.split("\\|");
+        String taskType = params[0].strip();
+        Boolean isCompleted = params[1].contains("1");
+        String description = params[2].strip();
+        Task task = null;
+
+        if (taskType.equals("T")) {
+            task = new Todo(description);
+        }
+
+        if (taskType.equals("D")) {
+            String by = params[3].strip();
+            task = new Deadline(description, by);
+
+        }
+
+        if (taskType.equals("E")) {
+            String from = params[3].strip();
+            String to = params[4].strip();
+            task = new Event(description, from, to);
+
+        }
+
+        if (isCompleted && task != null) {
+            task.markAsDone();
+        }
+
+        return task;
+    }
+
+    /**
+     * Returns the tasks from the text file.
+     * @return the task list from the text file
+     */
+    public TaskList load() {
+        TaskList tasks = new TaskList();
+
+        try {
+            boolean directoryExists = java.nio.file.Files.exists(dataPath);
```

Comment:
Perhaps this would be a clearer name?
```suggestion
            boolean doesDirectoryExist = java.nio.file.Files.exists(dataPath);
```